packages/keystone/src/index.ts

```diff
@@ -0,0 +1,230 @@
+      zeroBalanceAccounts = 0
+    }
+    index++
+  }
+  return accounts
+}
+
+function keystone({
+  customNetwork
+}: {
+  customNetwork?: CustomNetwork
+} = {}): WalletInit {
+  const getIcon = async () => (await import('./icon')).default
+  return () => {
+    let accounts: Account[] | undefined
+    return {
+      label: 'Keystone',
+      getIcon,
+      getInterface: async ({ EventEmitter, chains }: GetInterfaceHelpers) => {
+        const { providers } = await import('ethers')
+        const { default: AirGapedKeyring } = await import(
+          '@keystonehq/eth-keyring'
+        )
+        const { TransactionFactory: Transaction } = await import(
+          '@ethereumjs/tx'
+        )
+
+        const { default: Common, Hardfork } = await import('@ethereumjs/common')
+
+        const keyring = AirGapedKeyring.getEmptyKeyring()
+        await keyring.readKeyring()
+
+        const eventEmitter = new EventEmitter()
+
+        let currentChain: Chain = chains[0]
+        const scanAccounts = async ({
+          derivationPath,
+          chainId,
+          asset
+        }: ScanAccountsOptions): Promise<Account[]> => {
+          currentChain =
+            chains.find(({ id }: Chain) => id === chainId) ?? currentChain
+
+          const provider = new providers.JsonRpcProvider(currentChain.rpcUrl)
+          return generateAccounts(keyring, provider)
+        }
+
+        const getAccounts = async () => {
+          accounts = await accountSelect({
+            basePaths,
+            assets,
+            chains,
+            scanAccounts,
+            walletIcon: await getIcon(),
+            supportsCustomPath: false
+          })
+
+          if (accounts.length) {
+            eventEmitter.emit('accountsChanged', [accounts[0].address])
+          }
+
+          return accounts
+        }
+
+        const keystoneProvider = {}
+        const provider = createEIP1193Provider(keystoneProvider, {
+          eth_requestAccounts: async () => {
+            // Triggers the account select modal if no accounts have been selected
+            const accounts = await getAccounts()
+            if (accounts?.length === 0) {
+              throw new ProviderRpcError({
+                code: 4001,
+                message: 'User rejected the request.'
+              })
+            }
+            return [accounts[0]?.address]
+          },
+          eth_accounts: async () => {
+            return accounts?.[0]?.address ? [accounts[0].address] : []
+          },
+          eth_chainId: async () => {
+            return currentChain?.id ?? ''
+          },
+          eth_signTransaction: async ({ params: [transactionObject] }) => {
+            if (!accounts)
+              throw new Error(
+                'No account selected. Must call eth_requestAccounts first.'
+              )
+
+            const account =
+              accounts.find(
+                account => account.address === transactionObject?.from
```

Comment:
```suggestion
                account => account.address === transactionObject.from
```
You could also add a check that `transactionObject` exists at the top and throw an error if it does not as is required